### ISSUE TITLE
Missed subnetSpec.tags while specifying the cidrBlocks of subnet

### ISSUE THREAD

**Issue by lanfuchao** [pulumi/pulumi-awsx#1233]
While creating vpc with specified cidrBlocks of subnet, the subnet created will be not set the tags from SubnetArgs.

Expected "subnet-tag-public" and "subnet-tag-private" are disappeared
![image](https://github.com/pulumi/pulumi-awsx/assets/8337945/f0aa593a-24b9-4116-8156-63199c11304e)

## How to reproduce
```
package main

import (
	"github.com/pulumi/pulumi-awsx/sdk/v2/go/awsx/ec2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/tidbcloud/infra-cd-go/pkg/util"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := ec2.NewVpc(ctx, "test-vpc", &ec2.VpcArgs{
			AvailabilityZoneNames: []string{"us-west-2a"},
			NatGateways: &ec2.NatGatewayConfigurationArgs{
				Strategy: ec2.NatGatewayStrategySingle,
			},
			CidrBlock:      util.Ref("10.0.0.0/16"),
			SubnetStrategy: util.Ref(ec2.SubnetAllocationStrategyAuto),
			Tags: pulumi.StringMap{
				"vpc-tag": pulumi.String("true"),
			},
			SubnetSpecs: []ec2.SubnetSpecArgs{
				{
					Type:       ec2.SubnetTypePublic,
					Name:       pulumi.StringRef("public"),
					CidrBlocks: []string{"10.0.0.0/20"},
					Tags: pulumi.ToStringMap(map[string]string{
						"subnet-tag-public": "1",
					}),
				},
				{
					Type:       ec2.SubnetTypePrivate,
					Name:       pulumi.StringRef("private"),
					CidrBlocks: []string{"10.0.128.0/20"},
					Tags: pulumi.ToStringMap(map[string]string{
						"subnet-tag-private": "1",
					}),
				},
			},
		})
		return err
	})
}

```

## Possible issue 
Missed to return tags in getSubnetSpecsExplicit function
https://github.com/pulumi/pulumi-awsx/blob/9f18b3866d2a16a59d662384a0c745e2e5603463/awsx/ec2/subnetDistributorNew.ts#L154-L159

**Comment by mjeffryes** [comment]
thanks for reporting this @lanfuchao. I agree that's probably where the tags are being dropped.

**Comment by flostadler** [comment]
This was fixed as part of: https://github.com/pulumi/pulumi-awsx/pull/1288 and released in [v2.10.0](https://github.com/pulumi/pulumi-awsx/releases/tag/v2.10.0)
